www/perl-lib/SMT/RegistrationSharing.pm

```diff
@@ -11,12 +11,14 @@ use DBI qw(:sql_types);
 use File::Slurp;
 use File::Temp;
 use File::Touch;
+use File::stat;
+use Fcntl ':flock';
 use SMT::Utils;
 use WWW::Curl::Easy;
 use XML::LibXML;
 
-# A somewhat random number per thread avoid log and lock contention
-my $anyNumber = int(rand(1000));
+# pid of the parent apache process
+my $anyNumber = getppid();
```

Comment:
This will lead to data corruption, in Perl getpid() is always the same number in any thread of an application. Each thread will write to the replay log at different times and depending on buffering data will be interleaved and thus inconsistent and corrupted.

Reply:
`flock` was taking care of writing correctly, but I agree -- slurping the whole file into memory isn't a good idea. I've reverted those changes, for now at least.